### PR TITLE
Make DropDownButton width behavior consistent with another widgets

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: js/core/utils/drop_down.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import windowUtils from './window';
```
Comment (DokaRus) [inline js/core/utils/drop_down.js]: let's move this file to `js/ui/drop_down_editor`

File: js/ui/drop_down_button.js [hunk 2]
```diff
@@ -337,7 +331,7 @@ const DropDownButton = Widget.inherit({
                 show: { type: 'fade', duration: 0, from: 0, to: 1 },
                 hide: { type: 'fade', duration: 400, from: 1, to: 0 }
             },
-            width: 'auto',
+            width: getElementWidth.bind(null, this.$element()),
```
Comment (DokaRus) [inline js/ui/drop_down_button.js]:
```suggestion
            width: () => getElementWidth(this.$element()),
```

File: js/ui/drop_down_button.js [hunk 3]
```diff
@@ -414,6 +408,29 @@ const DropDownButton = Widget.inherit({
         });
     },
 
+    _popupOptionChanged: function(args) {
+        const options = Widget.getOptionsFromContainer(args);
+
+        this._setPopupOption('width', getElementWidth.bind(null, this.$element()));
```
Comment (DokaRus) [inline js/ui/drop_down_button.js]:
```suggestion
        this._setPopupOption('width', () => getElementWidth(this.$element()));
```

File: js/ui/drop_down_button.js [hunk 4]
```diff
@@ -414,6 +408,29 @@ const DropDownButton = Widget.inherit({
...
+        if(optionsKeys.indexOf('width') !== -1 || optionsKeys.indexOf('height') !== -1) {
+            this._dimensionChanged();
+        }
+    },
+
+    _dimensionChanged: function() {
+        const popupWidth = getPopupWidth(this.option('dropDownOptions.width'));
+
+        if(popupWidth === undefined) {
+            this._setPopupOption('width', getElementWidth.bind(null, this.$element()));
```
Comment (DokaRus) [inline js/ui/drop_down_button.js]:
```suggestion
            this._setPopupOption('width', () => getElementWidth(this.$element()));
```

File: js/core/utils/drop_down.js (hunk 1 above)
Comment (ksercs) [reply js/core/utils/drop_down.js]: fixed

File: js/ui/drop_down_button.js (hunk 2 above)
Comment (ksercs) [reply js/ui/drop_down_button.js]: fixed

File: js/ui/drop_down_button.js (hunk 3 above)
Comment (ksercs) [reply js/ui/drop_down_button.js]: fixed

File: js/ui/drop_down_button.js (hunk 4 above)
Comment (ksercs) [reply js/ui/drop_down_button.js]: fixed

File: js/ui/drop_down_button.js [hunk 5]
```diff
@@ -18,6 +17,7 @@ import { isPlainObject, isDefined } from '../core/utils/type';
 import { ensureDefined } from '../core/utils/common';
 import Guid from '../core/guid';
 import { format as formatMessage } from '../localization/message';
+import { getElementWidth, getPopupWidth } from './drop_down_editor/drop_down';
```
Comment (DokaRus) [inline js/ui/drop_down_button.js]:
```suggestion
import { getElementWidth, getPopupWidth } from './drop_down_editor/utils';
```

Comment (ksercs) [reply js/ui/drop_down_button.js]: fixed

**Review by DokaRus — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**